diffkemp/compare.py

```diff
@@ -19,225 +18,413 @@ class OutputDirExistsError(Exception):
+    @staticmethod
+    def _modules_exist(old_fun_desc, new_fun_desc):
+        return old_fun_desc.mod is not None and new_fun_desc.mod is not None
+
+    def _handle_missing_module(self, fun, group_dir,
+                               group_name, old_fun_desc, group_printed):
+        fun_result = Result(Result.Kind.UNKNOWN, fun, fun)
+        self.result.add_inner(fun_result)
+        group_printed = self._print_fun_result(fun_result, fun,
+                                               group_dir,
+                                               group_name,
+                                               old_fun_desc,
+                                               group_printed)
+        return group_printed
 
+    def _handle_fun_result(self, fun_result, fun, group_dir,
+                           group_name, old_fun_desc, group_printed):
+
+        if self.args.regex_filter is not None:
+            # Filter results by regex
+            self._filter_result_by_regex(self.regex_pattern,
+                                         fun_result)
+
+        self.result.add_inner(fun_result)
+
+        # Printing information about failures and
+        # non-equal functions.
+        if fun_result.kind in [Result.Kind.NOT_EQUAL,
+                               Result.Kind.UNKNOWN,
+                               Result.Kind.ERROR] \
+           or self.config.full_diff:
+            group_printed = self._print_fun_result(fun_result, fun,
+                                                   group_dir,
+                                                   group_name,
+                                                   old_fun_desc,
+                                                   group_printed)
+        return group_printed
 
-def _print_fun_result(fun_result, config, output_dir, fun, group_dir,
-                      group_name, compare_args, old_fun_desc, group_printed):
-    if fun_result.kind != Result.Kind.NOT_EQUAL and \
-       not config.full_diff:
-        # Print the group name if needed
-        if group_name is not None and not group_printed:
+    @staticmethod
+    def _filter_result_by_regex(pattern, fun_result):
+        for called_res in fun_result.inner.values():
+            if pattern.search(called_res.diff):
+                return
+        fun_result.kind = Result.Kind.EQUAL
+
+    def _print_fun_result(self, fun_result, fun, group_dir,
+                          group_name, old_fun_desc, group_printed):
+        if fun_result.kind != Result.Kind.NOT_EQUAL and \
+           not self.config.full_diff:
+            # Print the group name if needed
+            if group_name is not None and not group_printed:
+                print("{}:".format(group_name))
+                group_printed = True
+            print("{}: {}".format(fun, str(fun_result.kind)))
+            return group_printed
+
+        # Create the output directory if needed
+        if self.output_dir is not None:
+            self._ensure_dir_exists(self.output_dir)
```

Comment:
I would maybe inline the method in here:
```suggestion
            os.makedirs(path, exist_ok=True)
```

Reply:
I did not realize that dir can be `None` (for stdout output), in that case it probably makes sense to leave it in the method.